### ISSUE TITLE
DH parameter generation shall provide a progress bar

### ISSUE THREAD

**Issue by fpietrosanti** [globaleaks/Tor2web#277]
When automatic DH parameters generation starts, it take a long time but does not gives any feedback other than
"Generating HTTPS DH parameters (hold on, this may take a while!)"

This ticket is to provide an output/progress on the generation of DH parameters.

**Comment by evilaliv3** [comment]
This would require to pass a ```BN_GENCB``` callback to ```int DH_generate_parameters_ex(DH *, int, int, BN_GENCB *);```

if i'm not wrong this this is currently not possible in cryptography

@alex do you have a magic solution to suggest us?

**Comment by alex** [comment]
You are correct that cryptography doesn't presently offer an API for this. The easiest solution would probably be to do the DH param generation in a background thread.

**Comment by evilaliv3** [comment]
using openssl binary directly you mean? 

i think we would rather simply keep the current message "Generating HTTPS DH parameters (hold on, this may take a while!)" and postpone this ticket to further improvement whenever cryptography would allow it through ```ffi.callback```

thanks @alex

**Comment by fpietrosanti** [comment]
Just output a dot "." every 1s, until DH parameters has been generated. fixed!
